### ISSUE TITLE
ReferenceError: document is not defined

### ISSUE THREAD

**Issue by ondrejkralik** [fuzhenn/node-svg2img#43]
Hello,

I've got a problem with converting one specific SVG on **svg2img >= 0.8.0**
Probably this issue is connected with the upgrade to `canvg@3.0.6` in `svg2img@0.8.0` ;(

```
ReferenceError: document is not defined
    at Document.createCanvas (/Users/svg2img_example/node_modules/canvg/lib/index.js:7442:16)
    at MaskElement.apply (/Users/svg2img_example/node_modules/canvg/lib/index.js:7014:33)
    at PathElement.render (/Users/svg2img_example/node_modules/canvg/lib/index.js:2045:16)
    at /Users/svg2img_example/node_modules/canvg/lib/index.js:2096:15
    at Array.forEach (<anonymous>)
    at GElement.renderChildren (/Users/svg2img_example/node_modules/canvg/lib/index.js:2095:79)
    at GElement.render (/Users/svg2img_example/node_modules/canvg/lib/index.js:2057:14)
    at /Users/svg2img_example/node_modules/canvg/lib/index.js:2096:15
    at Array.forEach (<anonymous>)
    at GElement.renderChildren (/Users/svg2img_example/node_modules/canvg/lib/index.js:2095:79)
```

**Source code:**
```javascript
const fs = require('fs');
const svg2img = require('svg2img');

const svgString = `
<svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <defs>
    <polygon id="path-1" points="0 0.244013115 10.9193705 0.244013115 10.9193705 9.1730623 0 9.1730623"></polygon>
    <polygon id="path-3" points="0.0249180328 0.244039344 9.88065574 0.244039344 9.88065574 9.70491803 0.0249180328 9.70491803"></polygon>
  </defs>
  <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
    <g transform="translate(0.000000, 3.000000)">
      <g transform="translate(0.000000, 0.018308)">
        <mask id="mask-2" fill="white">
          <use xlink:href="#path-1"></use>
        </mask>
        <path d="M10.7848131,9.0075541 C10.1563541,8.23483279 9.4840918,7.36427541 8.39687869,7.30394754 L8.39425574,7.30368525 L8.2418623,7.30368525 L4.39687869,7.30368525 C3.04002623,7.30368525 2.52802623,5.65358689 2.55399344,4.54224262 C2.5773377,3.52191475 3.09353443,2.21909508 4.29091148,2.21909508 L8.88553443,2.21909508 L9.03792787,2.21909508 L9.03792787,0.244013115 L8.88553443,0.244013115 L4.29143607,0.244013115 L4.13904262,0.244013115 C1.62940328,0.244013115 0.0852721311,2.15483279 0.00343606557,4.5650623 C-0.0193836066,5.23968525 0.0682229508,5.92847213 0.284878689,6.56899672 C0.491829508,7.17961967 0.818911475,7.75063607 1.28238689,8.20283279 C1.96592787,8.86958689 2.83464918,9.1730623 3.78179672,9.1730623 L3.93445246,9.1730623 L10.766977,9.1730623 L10.9193705,9.1730623 L10.7848131,9.0075541 Z" fill="#293033" mask="url(#mask-2)"></path>
      </g>
      <g transform="translate(6.032787, 0.018308)">
        <mask id="mask-4" fill="white">
          <use xlink:href="#path-3"></use>
        </mask>
        <path d="M9.67134426,9.52535082 C8.45114754,8.49925246 7.45154098,7.4568918 7.11186885,5.86580984 C8.38714754,5.59643279 9.42819672,4.46672787 9.42977049,3.13033443 C9.43160656,1.5156459 8.09521311,0.244039344 6.48891803,0.244039344 L6.33363934,0.244039344 L3.47409836,0.244039344 L3.31881967,0.244039344 L3.32013115,2.21859672 L3.47514754,2.21859672 L5.84367213,2.21859672 C6.4052459,2.21859672 6.95868852,2.46830164 6.98177049,3.09649836 C6.97180328,3.35302295 6.90832787,3.60902295 6.71396721,3.78921967 C6.45560656,4.02817049 6.01154098,4.08036721 5.63068852,4.08561311 C5.58636066,4.08508852 5.54281967,4.08403934 5.50190164,4.08299016 C4.81101639,4.06305574 4.12380328,3.97230164 3.43029508,3.9581377 C3.34006557,3.95630164 3.24904918,3.95577705 3.15777049,3.95656393 C2.49862295,3.95079344 1.81219672,4.01400656 1.20498361,4.28207213 C0.78452459,4.46751475 0.43147541,4.73925246 0.148721311,5.10253115 L0.0249180328,5.26174426 L0.226622951,5.26620328 C1.44288525,5.29348197 2.51304918,5.57282623 3.51213115,6.27944918 C4.29744262,6.83446557 4.90544262,7.55577705 5.60813115,8.20076066 C6.75986885,9.25807213 7.86281967,9.72102295 9.44498361,9.70449836 C9.4984918,9.70502295 9.55252459,9.70502295 9.60708197,9.70449836 L9.88065574,9.70135082 L9.67134426,9.52535082 Z" fill="#0090D0" mask="url(#mask-4)"></path>
      </g>
    </g>
  </g>
</svg>
`;

svg2img(svgString, function(error, buffer) {
    if (error) {
        console.error(error);
        return;
    }

    fs.writeFileSync('foo1.png', buffer);
});
```

Thank you

**Comment by fuzhenn** [comment]
`at MaskElement.apply (/Users/svg2img_example/node_modules/canvg/lib/index.js:7014:33)`

Looks like caused by mask element's parsing in the svg.
It should be a bug of upstream canvg, it shouldn't use `document.createCanvas` in node environment.

Please report to [canvg issues](https://github.com/canvg/canvg/issues) to see if they can make a fix.

**Comment by ivands** [comment]
Got the same problem.

**Comment by FHGDev** [comment]
Hey does anyone have a fix? I'm getting the same issue.

**Comment by FHGDev** [comment]
Might just downgrade canvg

**Comment by bperel** [comment]
Created https://github.com/canvg/canvg/issues/1286

**Comment by gabelerner** [comment]
Hey canvg author here!

On https://github.com/fuzhenn/node-svg2img/blob/8995b92519380f0422ea316563f2c0bdac704341/index.js#L72 you're missing a key for `createCanvas` and that's why it's using document. The https://github.com/canvg/canvg/blob/22ad6475894f316c76cc2d15a1e768829428106a/src/presets/node.ts#L51 interface should be fully filled in. The easiest way to do this is to follow the presets in the node example on https://github.com/canvg/canvg#example

**Comment by bperel** [comment]
Thank you for the investigation @gabelerner . Is it something that could be done easily @fuzhenn ?

**Comment by bperel** [comment]
@fuzhenn Sorry for pinging, but do you think that you could help with this issue?

**Comment by fuzhenn** [comment]
should be fixed by #48
